### DISCUSSION TITLE
Allow to set pre-define clip length option for each column separately

### DISCUSSION THREAD

**Discussion by johnsamuelson** [helgoboss/helgobox#980] [Ideas]
Again hopefully a simple one but let's say you know in advance you want to record a loop of 8 bars but don't necessarily know that you'll be able to stop recording at the right point (your hands might be busy playing, and your feet might be busy doing other things too) - allowing a pre-defined loop length would mean you could record whenever you wanted, but would be safe in the knowledge that you wouldn't do some weird, horrible 9 bar phrase where you only meant to do an 8 bar one!

Currently this is achievable by dropping in empty MIDI regions of the correct length to the cells in question, but then you run into the overdub MIDI notes not displaying bug... plus it's just another layer of faff that no one needs!

Thanks!

**Comment by helgoboss** [comment]
Hi! You mean like this? https://www.youtube.com/watch?v=0PIjJlWtciw&t=469s

**Reply by johnsamuelson** [reply]
Ah - yes - but ideally per column if I were being picky? Thanks!

**Reply by helgoboss** [reply]
Could you outline the use case? What's the benefit of having it per column? I considered that already in the past but ultimately found that it's not too useful. Columns are mostly used to represent different instruments. But why would one always have a certain clip length for e.g. a bass guitar and another one for vocals?

I'm asking because this is not just the initial development effort, it's about usability. I would need to add the record-length indication to the column header, which clutters up the GUI. Right now, there's one setting, you can see it immediately when looking at the smart-record button.

**Reply by johnsamuelson** [reply]
I play live in an electronic / amplified hybrid setup, and each Reaper project is the master / guide for the song, taking care of things like click track, MIDI routing, MIDI looping and any playback and VSTi elements.

I handle MIDI in and out for 4 members of the band and all of them have different parts and different looping requirements; some might play a 4-bar phrase at the start of the song, but by the end of it have moved on to a 6-bar phrase. 

In one song for me, for example, it starts off:
- Loop guitar for 4 bars
- Loop guitar 2 for 4 bars
- Loop synth part for 2 bars 
- Loop synth part for 8 bars
- Loop synth part 2 for 8 bars
- Re-loop synth part 1 for 8 bars

The other guys are looping stuff via MIDI through me at the same time, quite often to different lengths (eg an SPD-SX sending MIDI looping a 4-bar drum part), so a global setting isn't always going to work for me.

Like I say, I can get around this by loading pre-defined MIDI clips of the correct length to each column or part, but I suppose it's just because of the complicated setup / system we run.

**Reply by jimmydeer** [reply]
I would find this useful as well for building tracks and live-often am playing different length parts.

Or it would be cool if right-clicking the transport to limit recording length had lengths there to click if you'd rather not put on columns.
